### DISCUSSION TITLE
tmp_dir and host_tmp_dir arguments for singularityOperator

### DISCUSSION THREAD

**Discussion by fafa92** [apache/airflow#31295] [General]
Hi,

I was wondering if airflow has arguments to be passed in for singularityOperator to read a specific folder from inside the container and write back to the host. I found a similar concept for dockeroperator with tmp_dir and host_tmp_dir arguments, but not for singularity. Any help will be appreciated, thanks!

**Comment by potiuk** [comment]
I guess this is a question to @vsoch

**Comment by vsoch** [comment]
@fafa92 try using volumes https://github.com/apache/airflow/blob/94128303e17412315aacd529d75a2ef549cce1f5/airflow/providers/singularity/operators/singularity.py#L78 

If you are just writing to a temporary location, unless your setup with Singularity forbids it, the temporary directories (and your home) should already be bound (and with write) and you shouldn't need to do anything. If the path it outside of those locations, and you want to, for example, bind a host directory to somewhere in the container (to give it write) you can do something like `["/home/me/mydata:/path/in/container"]`.
